src/org/opendatakit/briefcase/export/CsvMapper.java

```diff
@@ -0,0 +1,334 @@
+   * @param instanceId      the instance ID of the {@link Submission} instance that it's being exported
+   * @param workingDir      the {@link Path} to the working directory of the {@link Submission} instance that it's being exported
+   * @return a {@link List} of CSV compatible {@link String} lines
+   */
+  static String getRepeatCsvLine(Model groupModel, List<XmlElement> elements, boolean exportMedia, Path exportMediaPath, String instanceId, Path workingDir) {
+    StringBuilder sb = new StringBuilder();
+    elements.forEach(element -> {
+      String localId = element.getCurrentLocalId(instanceId);
+      StringBuilder sb2 = new StringBuilder();
+      groupModel.forEach(field -> getMapper(field).apply(
+          localId,
+          workingDir,
+          field,
+          element.findElement(field.getName()),
+          exportMedia,
+          exportMediaPath
+      ).forEach(pair -> sb2.append(",").append(encode(pair.right, pair.left.startsWith("meta") || pair.left.startsWith("SET-OF")))));
+
+      sb2.append(",").append(encode(element.getParentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getCurrentLocalId(instanceId), false));
+      sb2.append(",").append(encode(element.getGroupLocalId(instanceId), false));
+      sb.append(sb2.toString().substring(1)).append("\n");
+    });
+    return sb.toString();
+  }
+
+  private static FieldMapper getMapper(Model field) {
+    return Optional.ofNullable(mappers.get(field.getDataType()))
+        // If no mapper has been defined, we'll just output the text
+        .orElse(simpleMapper(CsvMapper::text));
+  }
+
+  private static List<Pair<String, String>> text(XmlElement element) {
+    return singletonList(Pair.of(
+        element.fqn(),
+        element.maybeValue().orElse(null)
+    ));
+  }
+
+  private static List<Pair<String, String>> date(XmlElement element) {
+    return formattedDate(element, getDateInstance());
+  }
+
+  private static List<Pair<String, String>> time(XmlElement element) {
+    return formattedDate(element, getTimeInstance());
+  }
+
+  private static List<Pair<String, String>> dateTime(XmlElement element) {
+    return formattedDate(element, getDateTimeInstance());
+  }
+
+  private static List<Pair<String, String>> geopoint(XmlElement element) {
+    String[] fields = Optional.ofNullable(element.getValue())
+        .map(v -> v.split(" "))
+        .orElseThrow(() -> new BriefcaseException("No value on geopoint field"));
+
+    List<Pair<String, String>> values = new ArrayList<>();
+    values.add(Pair.of(element.fqn() + "-Latitude", Optional.ofNullable(fields[0]).orElse(null)));
+    values.add(Pair.of(element.fqn() + "-Longitude", Optional.ofNullable(fields[1]).orElse(null)));
+    values.add(Pair.of(element.fqn() + "-Altitude", Optional.ofNullable(fields[2]).orElse(null)));
+    values.add(Pair.of(element.fqn() + "-Accuracy", Optional.ofNullable(fields[3]).orElse(null)));
+    return values;
```

Comment:
I didn’t like the duplication here so I tried my hand at Java 8 streams. Would you consider something like this?
```
    String[] tags = new String[] {"Latitude", "Longitude", "Altitude", "Accuracy"};
    return IntStream.range(0, tags.length).mapToObj(i ->
        Pair.of(element.fqn() + "-" + tags[i], Optional.ofNullable(fields[i]).orElse(null))
    ).collect(Collectors.toList());
```
![screen shot 2018-05-12 at 5 33 41 pm](https://user-images.githubusercontent.com/55099/39962664-b16c8b66-560a-11e8-8d60-391aca685dc0.png)

Reply:
Thanks for that, and for leading us into the past future with Java 8 functional features